LEGO1/lego/legoomni/src/race/legoracers.cpp

```diff
@@ -257,10 +275,61 @@ MxU32 LegoRaceCar::HandleSkeletonKicks(float p_param1)
 	return TRUE;
 }
 
-// STUB: LEGO1 0x100131f0
+// FUNCTION: LEGO1 0x100131f0
+// FUNCTION: BETA10 0x100cb88a
 void LegoRaceCar::VTable0x70(float p_float)
 {
-	// TODO
+	if (m_userNavFlag != 0 && (m_userState == LEGORACECAR_KICK1 || m_userState == LEGORACECAR_KICK2)) {
+		FUN_10012ff0(p_float);
+		return;
+	}
+
+	LegoCarRaceActor::VTable0x70(p_float);
+
+	if (m_userNavFlag != 0 && m_userState == LEGORACECAR_UNKNOWN_1) {
+		if (HandleSkeletonKicks(p_float)) {
+			return;
+		}
+	}
+
+	if (LegoCarRaceActor::m_unk0x0c == 1) {
+		FUN_1005d4b0();
+
+		if (m_userNavFlag == 0) {
+			FUN_10080590(p_float);
+			return;
+		}
+
+		float absoluteSpeed = abs(m_worldSpeed);
+		float maximumSpeed = NavController()->GetMaxLinearVel();
+		char buffer[200];
+
+		sprintf(buffer, "%g", absoluteSpeed / maximumSpeed);
+
+		VariableTable()->SetVariable(g_strSpeedCopy, buffer);
+
+		if (m_sound) {
+			// pitches up the engine sound based on the velocity
+			if (0.83 * maximumSpeed < absoluteSpeed) {
+				m_frequencyFactor = 1.9f;
+			}
+			else {
+				// this value seems to simulate RPM based on the gear
+				int gearRpmFactor = (int) (6.0 * absoluteSpeed) % 100;
+				m_frequencyFactor = gearRpmFactor / 80.0 + 0.7;
+			}
+		}
+
+		if (absoluteSpeed != 0.0f) {
+			g_unk0x100f0b88 = p_float;
+			g_unk0x100f0b8c = FALSE;
+		}
+
+		if (5000.0f < p_float - g_unk0x100f0b88 && !g_unk0x100f0b8c) {
```

Comment:
it would be better/more readable to flip the operands and the operator so that constant values appear on the right side of the test - should not affect match